core/types/json.go

```diff
@@ -44,6 +51,13 @@ func (b *hexBytes) UnmarshalJSON(input []byte) error {
 
 type hexBig big.Int
 
+func (b *hexBig) MarshalJSON() ([]byte, error) {
```

Comment:
Not part of your PR, but could you change line 41:
`if len(input) < 2 || input[0] != '0' || input[1] != 'x' {`
that it accepts '0X....' as well?